### PR TITLE
Improve forc documenter

### PR DESCRIPTION
**Pull request by binggh**
Partially addresses #1489 (1/2)

This PR is a general improvement on CI reporting of `forc-documenter`, after some feedback on how the CI instructions after a failed run can be hard to follow/understand.

Now it should report commands for which documentation is inconsistent and continue running, rather than terminating at the first failed command.

Example output:

```
   Compiling forc-documenter v0.0.0 (/Users/bh/Projects/fuel/sway/scripts/forc-documenter)
    Finished dev [unoptimized + debuginfo] target(s) in 1.19s
     Running `target/debug/forc-documenter write-docs --dry-run`
Running forc --help using forc 0.12.1

[✓] forc addr2line: documentation ok.
[x] forc build: documentation inconsistent!
[✓] forc clean: documentation ok.
[✓] forc completions: documentation ok.
[x] forc deploy: documentation inconsistent!
[✓] forc init: documentation ok.
[✓] forc json-abi: documentation ok.
[✓] forc parse-bytecode: documentation ok.
[x] forc plugins: documentation inconsistent!
[x] forc run: documentation inconsistent!
[✓] forc test: documentation ok.
[✓] forc update: documentation ok.
[x] index.md inconsistent!
[✓] SUMMARY.md ok.
Error: The Forc section of the Sway book needs to be updated.

Please run `cargo run --bin forc-documenter write-docs`. If you have made local changes to any forc native commands, please install forc from path first: `cargo install --path ./forc`, then run the command.

```

### CODE REVIEW

**Review by JoshuaBatty — DISMISSED** [review]
utACK, will be super handy. Thanks!

File: scripts/forc-documenter/src/checkers.rs [hunk 1]
```diff
@@ -20,31 +19,27 @@ pub fn is_options_line(line: &str) -> bool {
...
     if existing_index_contents == new_index_contents {
-        println!("index.md ok.");
+        println!("[✓] index.md ok.");
     } else {
-        return Err(anyhow!(
-            "index.md inconsistent - {}",
-            constants::RUN_WRITE_DOCS_MESSAGE
-        ));
+        eprintln!("[x] index.md inconsistent!");
+        return Err(anyhow!("index.md is inconsistent."));
```
Comment (mitchmindtree) [inline scripts/forc-documenter/src/checkers.rs:42]: Not important, but a hot-tip - you can use `bail!(...)` as an alternative to `return Err(anyhow!(...)));`

**Review by mitchmindtree — APPROVED** [review]
Nice

File: scripts/forc-documenter/src/checkers.rs (hunk 1 above)
Comment (binggh) [reply scripts/forc-documenter/src/checkers.rs:42]: ^thanks for the tip! I'll quickly change it

File: scripts/forc-documenter/src/main.rs [hunk 2]
```diff
@@ -220,7 +221,13 @@ fn write_docs(command: WriteDocsCommand) -> Result<()> {
             .expect("Failed to write to SUMMARY.md");
     }
 
+    if failed {
+        return Err(anyhow!(
+            "The Forc section of the Sway book needs to be updated. \n\nPlease run `cargo run --bin forc-documenter write-docs`. If you have made local changes to any forc native commands, please install forc from path first: `cargo install --path ./forc`, then run the command."
+        ));
```
Comment (mitchmindtree) [inline scripts/forc-documenter/src/main.rs:227]: Ah this could also be `bail!` - I missed this last time!

**Review by mitchmindtree — APPROVED** [review]
